c/parallel/src/transform.cu

```diff
@@ -0,0 +1,617 @@
+
+  const std::string input1_iterator_t =
+    input1_it.type == cccl_iterator_kind_t::CCCL_POINTER //
+      ? cccl_type_enum_to_name<input1_storage_t>(input1_it.value_type.type, true) //
+      : transform::get_input1_iterator_name();
+
+  const std::string input2_iterator_t =
+    input2_it.type == cccl_iterator_kind_t::CCCL_POINTER //
+      ? cccl_type_enum_to_name<input2_storage_t>(input2_it.value_type.type, true) //
+      : transform::get_input2_iterator_name();
+
+  const std::string output_iterator_t =
+    (output_it.type == cccl_iterator_kind_t::CCCL_POINTER //
+       ? cccl_type_enum_to_name<output_storage_t>(output_it.value_type.type, true) //
+       : transform::get_output_iterator_name());
+
+  std::string offset_t;
+  check(nvrtcGetTypeName<OffsetT>(&offset_t));
+
+  std::string transform_op_t;
+  check(nvrtcGetTypeName<op_wrapper>(&transform_op_t));
+
+  return std::format(
+    "cub::detail::transform::transform_kernel<{0}, {1}, {2}, {3}, {4}, {5}>",
+    chained_policy_t, // 0
+    offset_t, // 1
+    transform_op_t, // 2
+    output_iterator_t, // 3
+    input1_iterator_t, // 4
+    input2_iterator_t); // 5
+}
+
+template <auto* GetPolicy>
+struct dynamic_transform_policy_t
+{
+  using max_policy = dynamic_transform_policy_t;
+
+  template <typename F>
+  cudaError_t Invoke(int /*device_ptx_version*/, F& op)
+  {
+    return op.template Invoke<transform_runtime_tuning_policy>(GetPolicy());
+  }
+};
+
+struct transform_kernel_source
+{
+  cccl_device_transform_build_result_t& build;
+
+  CUkernel TransformKernel() const
+  {
+    return build.transform_kernel;
+  }
+
+  int LoadedBytesPerIteration()
+  {
+    return build.loaded_bytes_per_iteration;
+  }
+
+  template <typename It>
+  constexpr It MakeIteratorKernelArg(It it)
+  {
+    return it;
+  }
+};
+
+} // namespace transform
+
+CUresult cccl_device_unary_transform_build(
+  cccl_device_transform_build_result_t* build_ptr,
+  cccl_iterator_t input_it,
+  cccl_iterator_t output_it,
+  cccl_op_t op,
+  int cc_major,
+  int cc_minor,
+  const char* cub_path,
+  const char* thrust_path,
+  const char* libcudacxx_path,
+  const char* ctk_path)
+{
+  CUresult error = CUDA_SUCCESS;
+
+  try
+  {
+    const char* name = "test";
+
+    const int cc                 = cc_major * 10 + cc_minor;
+    const auto policy            = transform::get_policy();
+    const auto input_it_value_t  = cccl_type_enum_to_name<input_storage_t>(input_it.value_type.type);
+    const auto output_it_value_t = cccl_type_enum_to_name<output_storage_t>(output_it.value_type.type);
+    const auto offset_t          = cccl_type_enum_to_name(cccl_type_enum::CCCL_INT64);
+    const std::string input_iterator_src =
+      make_kernel_input_iterator(offset_t, "input_iterator_t", input_it_value_t, input_it);
+    const std::string output_iterator_src =
+      make_kernel_output_iterator(offset_t, "output_iterator_t", output_it_value_t, output_it);
```

Comment:
For consistency, instead of `"input_iterator_t"` maybe use `get_input_iterator_name()`? (Same comment for `output_iterator_t`)